qiita_db/processing_job.py

```diff
@@ -488,9 +490,34 @@ def get_resource_allocation_info(self):
                     AssertionError(
                         "Could not match %s to a resource allocation!" % name)
 
-            # [0] sending one element as execute_fetchflatten returns
-            # an array
-            return result[0]
+            allocation = result[0]
+
+            if ('{samples}' in allocation or '{columns}' in allocation or
+                    '{input_size}' in allocation):
+                samples, columns, input_size = self.shape
+                parts = []
+                for part in allocation.split():
+                    if ('{samples}' in part or '{columns}' in part or
+                            '{input_size}' in part):
+                        variable, value = part.split('=')
+                        error_msg = ('Obvious incorrect allocation. Please '
+                                     'contact qiita.help@gmail.com')
+                        if (('{samples}' in value and samples is None) or
+                                ('{columns}' in value and columns is None) or
+                                ('{input_size}' in value and input_size is
+                                 None)):
+                            self._set_error(error_msg)
+                            return 'Not valid'
```

Comment:
Would you mind elaborating a bit on what is happening here, I don't think I can follow. Adding some comments might help.

Reply:
Sure  ...